File: XamarinStore.iOS/Helpers/UIImageExtensions.cs
using System;
using MonoTouch.UIKit;
using System.Threading.Tasks;
using MonoTouch.CoreGraphics;
using System.Drawing;

namespace XamarinStore
{
	public static class UIImageExtensions
	{
		public static async Task LoadUrl(this UIImageView imageView, string url)
		{	
			if (string.IsNullOrEmpty (url))
				return;
			var progress = new UIActivityIndicatorView (UIActivityIndicatorViewStyle.WhiteLarge)
			{
				Center = new PointF(imageView.Bounds.GetMidX(), imageView.Bounds.GetMidY()),
			};
			imageView.AddSubview (progress);

		
			var t = FileCache.Download (url);
			if (t.IsCompleted) {
				imageView.Image = UIImage.FromFile(t.Result);
				progress.RemoveFromSuperview ();
				return;
			}
			progress.StartAnimating ();
			var image = UIImage.FromFile(await t);

			UIView.Animate (.3, 
				() => imageView.Image = image,
				() => {
					progress.StopAnimating ();
					progress.RemoveFromSuperview ();
				});
		}
	}
}

File: F#/XamarinStore.iOS/Helpers/UIImageExtensions.fs
namespace XamarinStore

open System
open MonoTouch.UIKit
open System.Threading.Tasks
open MonoTouch.CoreGraphics
open System.Drawing

module UIImageHelpers =
    type UIImageView with
        member this.LoadUrl url = async {
            if not (String.IsNullOrEmpty(url)) then
                let progress = new UIActivityIndicatorView (UIActivityIndicatorViewStyle.WhiteLarge, 
                                                            Center = new PointF(this.Bounds.GetMidX(), this.Bounds.GetMidY()))
                this.AddSubview (progress)

                progress.StartAnimating ()
                                                            
                let! t = FileCache.Download (url)
                let image = UIImage.FromFile(t)

                UIView.Animate (0.3, 
                    (fun () -> this.Image <- image),
                    fun () -> progress.StopAnimating ()
                              progress.RemoveFromSuperview () ) }